composer/trainer/trainer.py

```diff
@@ -226,32 +226,33 @@ def __init__(
         if not algorithms:
             algorithms = []
 
+        # some algorithms require specific settings
         self.backwards_create_graph = any(map(lambda x: x.backwards_create_graph, algorithms))
-
         find_unused_parameters = any(map(lambda x: x.find_unused_parameters, algorithms))
-
         self.find_unused_parameters = find_unused_parameters
 
+        # optionally initialize distributed backends
         if self.deepspeed_enabled:
             import deepspeed
             deepspeed.init_distributed()
-        else:
+        elif dist.get_world_size() > 1:
```

Comment:
Should we have a code path for our trainer to support dist initialization when the world size is 1? 

I'm thinking perhaps we can add a helper method to `composer.utils.dist` called something like `should_use_dist` that returns True if  a) all dist env variables are set, or b) the world size > 1. I'm thinking this code path could be helpful during development if someone wants to see how distributing wrapping affects a model / algorithm without having to run a multi-process test.

Reply:
Discovered that we need this code path in #389 for deepspeed.